icc-x-api/crypto/SecureDelegationsSecurityMetadataDecryptor.ts

```diff
@@ -106,10 +108,16 @@ export class SecureDelegationsSecurityMetadataDecryptor implements SecurityMetad
     if (!remainingDelegations.length) return { permissionsByDataOwnerId: accumulatedPermissions, hasUnknownAnonymousDataOwners }
     // 3. Attempt to identify the anonymous data owner of remaining delegations between us (or one of our parents) and an anonymous data owner
     const hierarchy = await this.dataOwnerApi.getCurrentDataOwnerHierarchyIds()
-    for (const delegationEntry of remainingDelegations) {
-      const delegation = delegationEntry[1]
+    const allHashes = remainingDelegations.flatMap(([hash, _]) => hash)
+    const encryptedExchangeDataIds = (await this.exchangeDataMap.getExchangeDataMapBatch(allHashes)).reduce((maps, current) => {
+      return {
+        ...maps,
+        [current.id]: current,
+      }
+    }, {} as { [hash: string]: ExchangeDataMap })
+    for (const [hash, delegation] of remainingDelegations) {
       if (hierarchy.some((x) => x === delegation.delegate || x === delegation.delegator)) {
-        const dataId = await this.secureDelegationsEncryption.decryptExchangeDataId(delegation)
+        const dataId = await this.secureDelegationsEncryption.decryptExchangeDataId(encryptedExchangeDataIds[hash].encryptedExchangeDataIds)
```

Comment:
Check also hash equivalences and assume that you may not find the exchange data map